ui/src/app/applications/components/application-details/application-resource-filter.tsx

```diff
@@ -0,0 +1,114 @@
+                const cur = update[prefix];
+                update[prefix] = `${cur ? cur + ',' : ''}${pair}`;
+            }
+        });
+        setGroupedFilters(update);
+        setLoading(false);
+    }, [resourceFilter, loading]);
+
+    useActionOnLargeWindow(() => setShown(true));
+
+    const setFilters = (prefix: string, values: string[]) => {
+        const groups = {...groupedFilters};
+        groups[prefix] = values.map(v => `${prefix}:${v}`).join(',');
+        let strings: string[] = [];
+        Object.keys(groups).forEach(g => {
+            strings = strings.concat(groups[g].split(',').filter(f => f !== ''));
+        });
+        onSetFilter(strings);
+    };
+
+    const ResourceFilter = (p: {label: string; prefix: string; options: string[]; field?: boolean; radio?: boolean}) => {
+        return loading ? (
+            <div>Loading...</div>
+        ) : (
+            <Filter
+                label={p.label}
+                selected={selectedFor(p.prefix)}
+                setSelected={v => setFilters(p.prefix, v)}
+                options={p.options.map(label => {
+                    return {label};
+                })}
+                field={!!p.field}
+                radio={!!p.radio}
+            />
+        );
+    };
+
+    // we need to include ones that might have been filter in other apps that do not apply to the current app,
+    // otherwise the user will not be able to clear them from this panel
+    const alreadyFilteredOn = (prefix: string) => resourceFilter.filter(f => f.startsWith(prefix + ':')).map(removePrefix(prefix));
+
+    const kinds = tree.nodes
+        .map(x => x.kind)
+        .concat(alreadyFilteredOn('kind'))
+        .filter(uniq)
+        .sort();
+    const namespaces = tree.nodes
+        .map(x => x.namespace)
+        .concat(alreadyFilteredOn('namespace'))
+        .filter(uniq)
+        .sort();
+
+    const selectedFor = (prefix: string) => {
+        return groupedFilters[prefix] ? groupedFilters[prefix].split(',').map(removePrefix(prefix)) : [];
+    };
+
+    return (
+        <>
+            <div className='applications-list__filters__title'>
+                FILTERS <i className='fa fa-filter' />
+                {pref.resourceFilter.length > 0 && (
+                    <ActionButton label={'CLEAR ALL'} action={() => onClearFilter()} style={{marginLeft: 'auto', fontSize: '12px', lineHeight: '5px', display: 'block'}} />
+                )}
+                <ActionButton
+                    label={!shown ? 'SHOW' : 'HIDE'}
+                    action={() => setShown(!shown)}
+                    style={{marginLeft: pref.resourceFilter.length > 0 ? '5px' : 'auto', fontSize: '12px', lineHeight: '5px', display: !shown && 'block'}}
+                />
+            </div>
+            {shown && (
+                <div className='applications-list__filters'>
+                    {ResourceFilter({label: 'OWNERSHIP', prefix: 'ownership', options: ['Owners', 'Owned']})}
```

Comment:
I think we should move the most frequently used filters to the top. Can you change order as following:

* KINDS
* SYNC STATUS
* OWNERSHIP
* CREATED WITHIN (rename it to AGE)